arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -1942,7 +1963,7 @@ void IResearchDataStore::afterTruncate(TRI_voc_tick_t tick,
     }
 
     // update reader
-    _dataStore._reader = reader;
+    std::atomic_store(&_dataStore._snapshot, reader);
```

Comment:
I think very unsafe to have _snapshot a public, I think it should be stored and loaded every time by separate function. We can avoid it only under commit mutex, because it also protects writes

Reply:
It is protected member, not intended to be used from outside. So what is unsafe here?

Reply:
It's probably not for this PR, but in general I believe we shouldn't have access to this members

Reply:
unsafe because you need to think where is safe to touch part of snapshot, and where is it unsafe.

I think everywhere except commit should use 
_snapshot.load() -> shared_ptr<Data>

And commit should use .store(Data)

Un noone shouldn't have access to _snapshot members

Reply:
That looks fearsible. The only place where we read only reader part  is metrics. But it would be no harm to load entire _snapshot. And that might be useful in future if we will include some data from storage snapshot into metrics